source/geod24/concurrency.d

```diff
@@ -1665,12 +1756,58 @@ public class Channel (T)
 
     public T receive ()
     {
+        this.mutex.lock();
+
+        if (this.closed)
+        {
+            this.mutex.unlock();
+            assert(0, "Channel is closed.");
+        }
+
         T msg;
```

Comment:
I would prefer if `T msg;` was declared inside of each `if` block separately. Because it will not be re-used. So:

```D
if (!this.sendq.empty)
{
    T msg;
    ...
}

if (!this.queue.empty)
{
    T msg = this.queue.front;
    ...
}
```

And so on..